### PR TITLE
Add pipeline job with pipeline component sample

### PR DESCRIPTION
**Pull request by brynn-code**
Add pipeline component sample
Fix typo in existing samples (1b_e2e_registered_components in cli not be fixed)

# PR into Azure/azureml-examples

## Checklist

I have:

- [x] read and followed the contributing guidelines

## Changes

-

fixes #

### CODE REVIEW

**Comment by brynn-code** [comment]
Add workflow.

File: cli/jobs/pipelines-with-components/pipeline_with_pipeline_component/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      training_data: ${{parent.inputs.pipeline_job_training_data}}
+      test_data: ${{parent.inputs.pipeline_job_test_data}}
+      max_epocs: 20
+      learning_rate: ${{parent.inputs.pipeline_job_training_learning_rate}}
+
+  compare:
+    type: command
+    component: file:./components/compare2.yaml
+    inputs:
+      model1: ${{parent.jobs.train_and_evaludate_model1.outputs.trained_model}}
```
Comment (wangchao1230) [inline cli/jobs/pipelines-with-components/pipeline_with_pipeline_component/pipeline.yml:52]: duplicate model1?

File: cli/jobs/pipelines-with-components/pipeline_with_pipeline_component/pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+$schema: https://azuremlschemas.azureedge.net/latest/pipelineJob.schema.json
```
Comment (wangchao1230) [inline cli/jobs/pipelines-with-components/pipeline_with_pipeline_component/pipeline.yml:1]: add an README.md for this sample

File: cli/assets/component/README.md [hunk 3]
```diff
@@ -21,7 +21,7 @@ This repository contains an example `YAML` file for creating `component` using A
 
 - To update a component that in workspace, execute following command. Currently only a few fields(description, display_name) support update:
 ```cli
-> az ml component update -f train.yml
+> az ml component update -n <component_name>
```
Comment (wangchao1230) [inline cli/assets/component/README.md:24]: add something like:
--set description='new description'

File: sdk/jobs/pipelines/1i_pipeline_with_pipeline_component/components/train_pipeline_component.yml [hunk 4]
```diff
@@ -0,0 +1,57 @@
+$schema: https://azuremlschemas.azureedge.net/latest/pipelineComponent.schema.json
```
Comment (wangchao1230) [inline sdk/jobs/pipelines/1i_pipeline_with_pipeline_component/components/train_pipeline_component.yml:1]: no need this for sdk sample

File: sdk/jobs/pipelines/1i_pipeline_with_pipeline_component/pipeline_with_pipeline_component.ipynb [hunk 5]
```diff
@@ -0,0 +1,307 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Build Pipeline Component with dsl\n",
```
Comment (wangchao1230) [inline sdk/jobs/pipelines/1i_pipeline_with_pipeline_component/pipeline_with_pipeline_component.ipynb:7]: Build Pipeline with Subpipeline (Pipeline Component)

File: sdk/jobs/pipelines/1i_pipeline_with_pipeline_component/pipeline_with_pipeline_component.ipynb [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+    "    training_learning_rate1=0.1,\n",
+    "    training_learning_rate2=0.01,\n",
+    "):\n",
+    "    # Create two training pipeline component with different learning rate\n",
+    "    train_and_evaludate_model1 = train_pipeline_component(\n",
+    "        training_input=training_input, \n",
+    "        test_input=test_input, \n",
+    "        training_learning_rate=training_learning_rate1\n",
+    "    )\n",
+    "    train_and_evaludate_model2 = train_pipeline_component(\n",
```
Comment (wangchao1230) [inline sdk/jobs/pipelines/1i_pipeline_with_pipeline_component/pipeline_with_pipeline_component.ipynb:200]: demo created pipeline component?

File: sdk/jobs/pipelines/1i_pipeline_with_pipeline_component/pipeline_with_pipeline_component.ipynb [hunk 7]
```diff
@@ -0,0 +1,307 @@
...
+    "        \"best_result\": compare2_models.outputs.best_result,\n",
+    "    }\n",
+    "\n",
+    "\n",
+    "pipeline_job = pipeline_with_pipeline_component(\n",
+    "    training_input=Input(type=\"uri_folder\", path=parent_dir + \"/data/\"),\n",
+    "    test_input=Input(type=\"uri_folder\", path=parent_dir + \"/data/\"),\n",
+    ")\n",
+    "\n",
+    "# set pipeline level compute\n",
```
Comment (wangchao1230) [inline sdk/jobs/pipelines/1i_pipeline_with_pipeline_component/pipeline_with_pipeline_component.ipynb:225]: set pipeline level settings: default compute

File: cli/assets/component/train_pipeline_component.yml [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+  learning_rate_schedule:
+    type: string
+    default: 'time-based'
+
+outputs: 
+  trained_model:
+    type: uri_folder
+
+jobs:
+  train_job:
```
Comment (wangchao1230) [inline cli/assets/component/train_pipeline_component.yml:24]: at least two nodes? 

Maybe add a ls comamnd node

File: cli/jobs/pipelines-with-components/pipeline_with_pipeline_component/components/train_src/train.yml [hunk 9]
```diff
@@ -0,0 +1,31 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandComponent.schema.json
```
Comment (wangchao1230) [inline cli/jobs/pipelines-with-components/pipeline_with_pipeline_component/components/train_src/train.yml:1]: train_src -> train  in this case

File: cli/jobs/pipelines-with-components/pipeline_with_pipeline_component/components/train_src/train.yml [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+type: command
+
+name: train_model
+display_name: Train Model
+description: A dummy training component
+version: 0.0.1
```
Comment (wangchao1230) [inline cli/jobs/pipelines-with-components/pipeline_with_pipeline_component/components/train_src/train.yml:7]: version could ommitted?

File: sdk/jobs/pipelines/1f_pipeline_with_pipeline_component/pipeline_with_pipeline_component.ipynb [hunk 11]
```diff
@@ -0,0 +1,325 @@
...
+    "**Requirements** - In order to benefit from this tutorial, you will need:\n",
+    "- A basic understanding of Machine Learning\n",
+    "- An Azure account with an active subscription - [Create an account for free](https://azure.microsoft.com/free/?WT.mc_id=A261C142F)\n",
+    "- An Azure ML workspace with computer cluster - [Configure workspace](../../configuration.ipynb)\n",
+    "- A python environment\n",
+    "- Installed Azure Machine Learning Python SDK v2 - [install instructions](../../../README.md) - check the getting started section\n",
+    "\n",
+    "**Learning Objectives** - By the end of this tutorial, you should be able to:\n",
+    "- Connect to your AML workspace from the Python SDK\n",
+    "- Define and load `CommandComponent` from YAML\n",
```
Comment (wangchao1230) [inline sdk/jobs/pipelines/1f_pipeline_with_pipeline_component/pipeline_with_pipeline_component.ipynb:18]: we can remove this line:  "- Define and load `CommandComponent` from YAML\n",

File: cli/jobs/pipelines-with-components/pipeline_with_pipeline_component/README.md [hunk 12]
```diff
@@ -0,0 +1 @@
+This is a 3 step dummy pipeline job. There are 2 pipeline nodes in the job, each of the pipeline node uses locally defined components - train, score and eval. You need to edit the compute cluster in the defaults section and run the `az ml job create --file pipeline.yml` to submit the pipeline job.
```
Comment (wangchao1230) [inline cli/jobs/pipelines-with-components/pipeline_with_pipeline_component/README.md:1]: 3 step -> 3 nodes

let's using consitent wording here

File: cli/jobs/pipelines/automl/cli-automl-classification-task-bankmarketing-pipeline/classification-task-bankmarketing-pipeline.yml [hunk 13]
```diff
@@ -25,21 +25,22 @@ jobs:
     limits:
       max_trials: 1
       max_concurrent_trials: 1
-    target_column_name: y
+    target_column_name: "y"
```
Comment (wangchao1230) [inline cli/jobs/pipelines/automl/cli-automl-classification-task-bankmarketing-pipeline/classification-task-bankmarketing-pipeline.yml:28]: why do we have automl changes in this PR

Comment (brynn-code) [reply cli/jobs/pipelines/automl/cli-automl-classification-task-bankmarketing-pipeline/classification-task-bankmarketing-pipeline.yml:28]: not sure, I'll revert this

**Review by D-W- — APPROVED** [review]

**Review by YingChen1996 — APPROVED** [review]
